### PR TITLE
chore: refactor component with user id service

### PR DESCRIPTION
**Pull request by dydome**
related to https://github.com/SAP/spartacus/pull/11358 code review comments

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts [hunk 1]
```diff
@@ -182,10 +185,10 @@ export class CartItemListComponent {
             value.entryNumber,
             value.quantity
           );
-        } else if (value && this.cart?.cartId && this.cart?.userId) {
+        } else if (value && this.cartId && userId) {
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: The longer I look it seems like the takeUserId + subscription would be better solution here.

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts [hunk 2]
```diff
@@ -89,6 +95,7 @@ export class CartItemListComponent {
   constructor(
     protected activeCartService: ActiveCartService,
     protected selectiveCartService: SelectiveCartService,
+    protected userIdService: UserIdService,
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: In this case `userIdService` should also be optional here like `featureConfigService` and `multiCartService`

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts (hunk 1 above)
Comment (bgambocjaviniar) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: fixed

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts (hunk 2 above)
Comment (bgambocjaviniar) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: fixed + removed unused import in the saved cart list for the userIdService

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts [hunk 3]
```diff
@@ -74,13 +84,15 @@ export class CartItemListComponent {
   constructor(
     activeCartService: ActiveCartService,
     selectiveCartService: SelectiveCartService,
+    userIdService: UserIdService,
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: It should not be present in this constructor variant and it should be after `featureConfigService`

Comment (marlass) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: Also don't forget to update deprecation ts docs

**Review by marlass — CHANGES_REQUESTED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**